fuzz/fuzz_targets/differential.rs

```diff
@@ -613,31 +620,43 @@ impl FuzzContext {
     }
 }
 
-fuzz_target!(|data: &[u8]| {
-    let Ok(mut smith_module) = arbitrary_exec_module(data) else {
-        return;
-    };
-    // Note: We cannot use built-in fuel metering of the different engines since that
-    //       would introduce unwanted non-determinism with respect to fuzz testing.
-    let Ok(_) = smith_module.ensure_termination(1_000 /* fuel */) else {
-        return;
-    };
-    let wasm = smith_module.to_bytes();
-    let Some(wasmi_register) = <WasmiRegister as DifferentialTarget>::setup(&wasm[..]) else {
-        return;
-    };
-    let Some(wasmi_stack) = <WasmiStack as DifferentialTarget>::setup(&wasm[..]) else {
-        panic!("wasmi (register) succeeded to create Context while wasmi (stack) failed");
-    };
-    let exports = wasmi_register.exports();
-    let mut context = FuzzContext {
-        wasm,
-        wasmi_register,
-        wasmi_stack,
-        exports,
-    };
-    context.run();
-});
+fn main() {
+    loop {
+        fuzz!(|seed: &[u8]| {
+            let mut unstructured = Unstructured::new(seed);
+            let Ok(mut smith_module) =
+                arbitrary_config(&mut unstructured).and_then(|mut config| {
```

Comment:
Could this be dis-entangled by removing the `and_then` and simply pulling out the `let else` here?

Reply:
The `let else` is here to silently fail in case wasm_smith encounters any errors which we don't want to show up during fuzzing. Not sure what you mean about removing `and_then`, splitting this statement into two `let`s?

Reply:
I meant something like this:
```rust
let Ok(config) = arbitrary_config(&mut unstructured) else {
    return;
}
config.reference_types_enabled = false;
config.tail_call_enabled = false;
let Ok(mut smith_module) = wasm_smith::Module::new(config, &mut unstructured) else {
    return;
};
```
However, this is a bit nit-picky tbh.